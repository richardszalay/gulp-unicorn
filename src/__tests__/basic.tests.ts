import { PluginEnvironment, createCompiler, createMockEnvironment } from "./helpers";

import * as webpack from "webpack";
import { RainbowPlugin } from "..";
import * as tapable from "tapable";

import { getUuid, enqueueUuid, getDate, setDate } from "./mocks";

jest.mock("uuid/v4", () => () => getUuid());

const { AsyncSeriesHook } = tapable;

describe("default options", () => {

    it("creates item if it doesn't exist", async () => {
        const compilation = createMockEnvironment("parent1");

        compilation.assets["test.js"] = {
            source: () => "function foo(bar) { return bar; }"
        };

        enqueueUuid("aaa", "bbb");
        Date.now = () => new Date("2018-01-02T03:04:05Z").valueOf();

        const promise = new Promise((res, rej) => new RainbowPlugin().emit(compilation, res));

        await expect(promise.then(() => compilation.assets["test.yml"].source().toString()))
            .resolves.toMatchSnapshot("new");

        await expect(promise.then(() => compilation.assets["test.js"]))
            .resolves.toBeUndefined();
    });

    it("updates item if it already exists", async () => {
        const compilation = createMockEnvironment("parent1");

        compilation.assets["child1.js"] = {
            source: () => "function foo(bar) { return bar; }"
        };

        enqueueUuid("aaa", "bbb");
        Date.now = () => new Date("2018-01-02T03:04:05Z").valueOf();

        const promise = new Promise((res, rej) => new RainbowPlugin().emit(compilation, res));

        await expect(promise.then(() => compilation.assets["child1.yml"].source().toString()))
            .resolves.toMatchSnapshot("exists");

        await expect(promise.then(() => compilation.assets["child1.js"]))
            .resolves.toBeUndefined();
    });
});
