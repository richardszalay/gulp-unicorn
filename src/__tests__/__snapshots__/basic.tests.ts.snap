// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default options creates item if it doesn't exist: new 1`] = `
"---
ID: aaa
Parent: parent1-id
Template: 962b53c4-f93b-4df9-9821-415c867b8903
Path: /content/Parent1/test
DB: master
SharedFields:
  - ID: 40e50ed9-ba07-4702-992e-a912738d32dc
    Hint: Blob
    BlobID: bbb
    Value: ZnVuY3Rpb24gZm9vKGJhcikgeyByZXR1cm4gYmFyOyB9
  - ID: 6954b7c7-2487-423f-8600-436cb3b6dc0e
    Hint: Size
    Value: 33
  - ID: 6f47a0a5-9c94-4b48-abeb-42d38def6054
    Hint: Mime Type
    Value: application/x-javascript
  - ID: ba3f86a2-4a1c-4d78-b63d-91c2779c1b5e
    Hint: __Sortorder
    Value: 10
  - ID: c06867fe-9a43-4c7d-b739-48780492d06f
    Hint: Extension
    Value: js
Languages:
  - Language: en
    Versions:
      - Version: 1
        Fields:
          - ID: 25bed78c-4957-4165-998a-ca1b52f67497
            Hint: __Created
            Value: 20180102T030405Z
          - ID: 5dd74568-4d4b-44c1-b513-0af5f4cda34f
            Hint: __Created by
            Value: sitecore\\\\admin
"
`;

exports[`default options updates item if it already exists: exists 1`] = `
"---
ID: child-id
Path: /content/Parent1/child1
Parent: parent1-id
SharedFields:
  - ID: 40e50ed9-ba07-4702-992e-a912738d32dc
    Hint: Blob
    BlobID: aaa
    Value: ZnVuY3Rpb24gZm9vKGJhcikgeyByZXR1cm4gYmFyOyB9
  - ID: 6954b7c7-2487-423f-8600-436cb3b6dc0e
    Hint: Size
    Value: 33
  - ID: 6f47a0a5-9c94-4b48-abeb-42d38def6054
    Hint: Mime Type
    Value: application/x-javascript
  - ID: ba3f86a2-4a1c-4d78-b63d-91c2779c1b5e
    Hint: __Sortorder
    Value: 10
  - ID: c06867fe-9a43-4c7d-b739-48780492d06f
    Hint: Extension
    Value: js
"
`;
